quinn-udp/src/unix.rs

```diff
@@ -690,7 +816,18 @@ mod gso {
     }
 }
 
-#[cfg(not(target_os = "linux"))]
+#[cfg(any(target_os = "macos", target_os = "ios"))]
+mod gso {
+    use super::*;
+
+    pub(super) fn max_gso_segments() -> usize {
+        BATCH_SIZE
+    }
```

Comment:
I think it is worth documenting here, that UDP Datagram segmentation is not offloaded to the OS or even better the NIC, but instead done here in user space in `fn send()` and then passed to the OS as individual `iovec`s.